### PR TITLE
Try: Display the site name in the top bar in Appearance > Editor

### PR DESCRIPTION
**Pull request by carolinan**
## Description

The site title, as a link to the homepage, should be visible in the top bar when you open the editor via
Appearance > Editor.

For https://github.com/WordPress/gutenberg/issues/36207

## How has this been tested?

Manually by opening the editor via Appearance > Editor.
 
## Types of changes

Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by carolinan** [comment]
How we handle the failing tests feels like a bigger issue than this single pull request.

It depends on removing the test(s?) for the removed navigation sidebar, and how the the template detail modal should be handled.

File: packages/edit-site/src/components/header/index.js [hunk 1]
```diff
@@ -103,6 +109,31 @@ export default function Header( {
 
 	const isFocusMode = templateType === 'wp_template_part';
 
+	let displayTitle;
+
+	if ( entityTitle === 'Index' || entityTitle === 'Home' ) {
```
Comment (carolinan) [inline packages/edit-site/src/components/header/index.js:114]: Gut says this is not correct and depends on what should be displayed when the editor is opened via `Appearance >templates` / `Appearance >template parts`

Comment (jameskoster) [reply packages/edit-site/src/components/header/index.js:114]: > Gut says this is not correct and depends on what should be displayed when the editor is opened via Appearance >templates / Appearance >template parts

When accessing a template via Appearance > Templates, it seems fair to continue displaying the template name.

Comment (carolinan) [reply packages/edit-site/src/components/header/index.js:114]: Except that path does not open in the Site Editor but the block editor. That still needs to be solved.

Comment (jameskoster) [reply packages/edit-site/src/components/header/index.js:114]: Indeed 🙈 https://github.com/WordPress/gutenberg/issues/35990

**Pull request closed without merging**